3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -1,29 +1,22 @@
 'use strict';
-/*------------------------------------------------------------------------------
-Full description at: https://github.com/HackYourFuture/Homework/tree/main/3-UsingAPIs/Week1#exercise-1-john-who
 
-Rewrite this function, but replace the callback syntax with the Promise syntax:
-- Have the `getAnonName` function return a `new Promise`.
-- If the Promise `resolves`, pass the full name as an argument to resolve with.
-- If the Promise `rejects`, pass an error as the argument to reject with: "You 
-  didn't pass in a first name!"
-------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
-
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
-};
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+      }else{
```

Comment:
It is customary to put spaces after a closing brace and before and opening brace, like so:

```suggestion
      } else {
```

Normally the Prettier VSCode extension takes care of this but it looks like it is not set up correctly in your VSCode. I will come back on how to fix that later.